### PR TITLE
feat: Edit Note

### PR DESCRIPTION
**Pull request by horsemaker**

### CODE REVIEW

File: src/backend/controllers/NotesController.js [hunk 1]
```diff
@@ -46,9 +46,14 @@ export const createNoteHandler = function (schema, request) {
...
-      user.notes.push({ ...note, _id: uuid(), tags: [] });
+      user.notes.push({
+        ...note,
+        _id: uuid(),
+        tags: [],
+        createdAt: new Date(),
+      });
     } else {
-      user.notes.push({ ...note, _id: uuid() });
+      user.notes.push({ ...note, _id: uuid(), createdAt: new Date() });
```
Comment (vijaytembugade) [inline src/backend/controllers/NotesController.js:56]: Thanks for giving idea about this. It gives createdAt by default.

File: src/frontend/components/NoteCard/NoteCard.js [hunk 2]
```diff
@@ -166,33 +180,47 @@ export const NoteCard = ({ note }) => {
...
           ))}
         {pathname === "/trash" && (
-          <button onClick={restoreFromTrash}>
+          <button className="btn-hover" onClick={restoreFromTrash}>
             <span className="material-icons">restore_from_trash</span>
           </button>
         )}
-        {pathname === "/notes" ? (
-          <button onClick={moveToTrash}>
+        {matchPath("/notes/*", pathname) && (
```
Comment (vijaytembugade) [inline src/frontend/components/NoteCard/NoteCard.js:210]: Good that you are using `matchpath`

File: src/frontend/components/ReactQuillEditor/ReactQuillEditor.css [hunk 3]
```diff
@@ -74,8 +74,15 @@
...
   font-size: 1rem;
   padding: 0;
+  max-height: 50vh;
+  -ms-overflow-style: none;
+  scrollbar-width: none;
 }
 
+/* .ql-editor::-webkit-scrollbar {
+  display: none;
+} */
```
Comment (vijaytembugade) [inline src/frontend/components/ReactQuillEditor/ReactQuillEditor.css]: remove this commented code!

**Review by omgaikwad — APPROVED** [review]
Clean code, good file structure, great that you made smaller reusable components, you are good to go!

**Review by vijaytembugade — APPROVED** [review]
Good work. Edit working very smooth. 
image

File: src/frontend/components/EditorModal/EditorModal.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+export const EditorModal = () => {
+  const textEditorRef = useRef();
+  const navigate = useNavigate();
+  const { noteId } = useParams();
+
+  const { notes } = useNotes();
+  const note = notes.find((note) => note._id === noteId);
+
+  useOnClickOutside(textEditorRef, () => navigate("/notes"));
+  useLockBodyScroll(true);
```
Comment (t007rushi) [inline src/frontend/components/EditorModal/EditorModal.js:17]: great use of hooks

File: src/frontend/components/ReactQuillEditor/ReactQuillEditor.css [hunk 5]
```diff
@@ -74,8 +74,15 @@
...
+
```
Comment (t007rushi) [inline src/frontend/components/ReactQuillEditor/ReactQuillEditor.css]: comments to remove

**Review by t007rushi — APPROVED** [review]
Nice Work Here

**Pull request merged**